core-java-modules/core-java-streams-4/src/test/java/splitting/StreamSplittingUnitTest.java

```diff
@@ -0,0 +1,86 @@
+package splitting;
+
+import com.google.common.collect.Lists;
+import org.junit.Assert;
+import org.junit.Test;
+import rx.Observable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StreamSplittingUnitTest {
+    // given
+    private final static List<Article> articles = Lists.newArrayList(
+            new Article("Baeldung", true),
+            new Article("Baeldung", false),
+            new Article("Programming Daily", false),
+            new Article("The Code", false)
+    );
+
+    @Test
+    public void givenListOfArticles_shouldSplitListInTwoCategories_usingPartitioningBy() {
+        // when
+        Map<Boolean, List<Article>> groupedArticles = articles.stream().collect(Collectors.partitioningBy(a -> a.target.equals("Baeldung")));
+
+        // then
+        Assert.assertEquals(2, groupedArticles.get(true).size());
```

Comment:
Let's static import the `assertEquals` - though to be honest, I'd rather use assertJ here - `assertThat(groupedArticles.get(true)).hasSize(2)`

On top of that, there's a question around whether a map of `Boolean` to `List<Article>` is a nice thing to have or not.

And if we added getters or `equals`/`hashCode` to `Article`, we could actually compare the precise articles that were split, rather than just the count

Reply:
Normally I also prefer AssertJ but from what I've seen in the Baeldung codebase JUnit assertions are more prevalent so I opted for consistency.
I added equals and hashCode and improved tests accordingly.